packages/xstate-solid/src/useSelector.ts

```diff
@@ -0,0 +1,57 @@
+import type { ActorRef, Subscribable } from 'xstate';
+import { defaultGetSnapshot } from './useActor';
+import type { Accessor } from 'solid-js';
+import { createEffect, createMemo, on, onCleanup } from 'solid-js';
+import { deepClone } from './utils';
+import { createStoreSignal } from './createStoreSignal';
+
+const defaultCompare = (a, b) => a === b;
+
+export function useSelector<
+  TActor extends ActorRef<any>,
+  T,
+  TEmitted = TActor extends Subscribable<infer Emitted> ? Emitted : never
+>(
+  actor: Accessor<TActor> | TActor,
+  selector: (emitted: TEmitted) => T,
+  compare: (a: T, b: T) => boolean = defaultCompare,
+  getSnapshot: (a: TActor) => TEmitted = defaultGetSnapshot
+): Accessor<T> {
+  const actorMemo = createMemo<TActor>(
+    typeof actor === 'function' ? actor : () => actor
+  );
+
+  // Deep clone to prevent mutating original snapshot
+  const getActorSnapshot = (snapshotActor: TActor): T =>
+    deepClone(selector(getSnapshot(snapshotActor)));
+
+  const [selected, update] = createStoreSignal<T, T>(
+    actorMemo,
+    getActorSnapshot
+  );
+
+  const guardedUpdate = (nextSelected: TEmitted) => {
+    if (!compare(selected(), selector(nextSelected))) {
+      update(selector(nextSelected));
+    }
+  };
```

Comment:
```suggestion
  const guardedUpdate = (emitted: TEmitted) => {
    const next = selector(emitted);
    if (!compare(selected(), next)) {
      update(next);
    }
  };
```

I wonder though if the `compare`-related logic is needed in the face of fine-grained reactivity. Have you seen any other Solid-related libraries doing this?